test/api_access_test.rb

```diff
@@ -0,0 +1,132 @@
+    refute_equal unauthenticated_scopes, authenticated_read_scopes
+    refute_equal unauthenticated_scopes, authenticated_write_scopes
+  end
+
+  def test_scopes_includes_is_truthy_for_same_scopes
+    scopes = ShopifyAPI::ApiAccess.new(%w(read_orders))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(read_orders))
+    assert scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_falsy_for_different_scopes
+    scopes = ShopifyAPI::ApiAccess.new(%w(read_orders))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(read_products))
+    refute scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_truthy_for_read_when_the_set_has_read_write
+    scopes = ShopifyAPI::ApiAccess.new(%w(write_products))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(read_products))
+    assert scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_truthy_for_read_when_the_set_has_read_write_for_that_resource_and_others
+    scopes = ShopifyAPI::ApiAccess.new(%w(write_products, write_orders))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(read_orders))
+    assert scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_truthy_for_write_when_the_set_has_read_write_for_that_resource_and_others
+    scopes = ShopifyAPI::ApiAccess.new(%w(write_products, write_orders))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(write_orders))
+    assert scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_truthy_for_subset_of_scopes
+    scopes = ShopifyAPI::ApiAccess.new(%w(write_products write_orders write_customers))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(write_orders read_products))
+    assert scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_falsy_for_sets_of_scopes_that_have_no_common_elements
+    scopes = ShopifyAPI::ApiAccess.new(%w(write_products write_orders write_customers))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(write_images read_content))
+    refute scopes.covers?(other_scopes)
+  end
+
+  def test_covers_is_falsy_for_sets_of_scopes_that_have_only_some_common_access
+    scopes = ShopifyAPI::ApiAccess.new(%w(write_products write_orders write_customers))
+    other_scopes = ShopifyAPI::ApiAccess.new(%w(write_products read_content))
+    refute scopes.covers?(other_scopes)
+  end
+
+  def test_duplicate_scopes_resolve_to_one_scope
+    scopes = ShopifyAPI::ApiAccess.new(%w(read_orders read_orders read_orders read_orders))
+    equivalent_scopes = ShopifyAPI::ApiAccess.new(%w(read_orders))
+    assert_equal equivalent_scopes, scopes
+  end
+
+  def test_to_s_outputs_scopes_as_a_comma_separated_list_without_implied_read_scopes
+    serialized_scopes = "read_products,write_orders"
+    expected_scopes = ShopifyAPI::ApiAccess.new(%w(read_products read_orders write_orders))
+    assert_equal expected_scopes.to_s, serialized_scopes
+  end
+
+  def test_to_a_outputs_scopes_as_an_array_of_strings_without_implied_read_scopes
+    serialized_scopes = %w(read_products write_orders)
```

Comment:
Should this be a string?

Reply:
It can either be a string or an array of scopes so that we can support two types of inputs when instantiating ApiAccess.

Reply:
I was just wondering if it made sense for `serialized_scopes` to be an array - I thought the tests was slightly off in that but it seems like it's just the variable name that is ''''wrong''''. No reason to change it 🙂